### PR TITLE
Update DCCRG neighbor definitions, decrease neighborhood size by 1.

### PR DESCRIPTION
**Pull request by ursg**
After fixing a small number of bugs that were left, I suppose this is now ready for testing:

This PR updates the DCCRG submodel to point at my branch, in which the definition of what neighbor is has been changed. As a result, we no longer need any extra speculative neighborhood sizes to make sure we capture all relevant neighbors at AMR interfaces, and can run with a neighborhood_size that is one unit smaller.

This results in smaller induced refinement regions, less communication overhead and thus, hopefully, an overall speedup.

After running a number of testpackage tests by hand, I'm now keen to see if this runs the whole package, or whether there are still issues to iron out.

Note: This PR also updates all makefiles to build with c++20. Alternatively, the erase_if implementation in dccrg needs to be replaced with a backwards-compatible erase(remove) idiom.

### CODE REVIEW

File: vlasovsolver/cpu_trans_map.cpp [hunk 1]
```diff
@@ -699,12 +703,22 @@ void update_remote_mapping_contribution(
             //2) is remote cell, 3) if the source cell in center was
             //translated
             ccell->neighbor_block_data[0] = pcell->get_data(popID);
+
+            for(unsigned int cell = 0; cell<VELOCITY_BLOCK_LENGTH * pcell->get_number_of_velocity_blocks(popID); ++cell) {
```
Comment (markusbattarbee) [inline vlasovsolver/cpu_trans_map.cpp:707]: Put NaN checks inside debug ifdefs?

File: vlasovsolver/cpu_trans_map.cpp [hunk 2]
```diff
@@ -734,14 +748,13 @@ void update_remote_mapping_contribution(
       break;
    }
    
-#pragma omp parallel
```
Comment (markusbattarbee) [inline vlasovsolver/cpu_trans_map.cpp:737]: was this removed by accident?

File: vlasovsolver/cpu_trans_map.cpp [hunk 3]
```diff
@@ -674,23 +657,44 @@ void update_remote_mapping_contribution(
       CellID p_ngbr = INVALID_CELLID;
       CellID m_ngbr = INVALID_CELLID;
 
-      for (const auto& [neighbor, dir] : mpiGrid.get_face_neighbors_of(local_cells[c])) {
+      //if(local_cells[c] == 4 || local_cells[c] == 5) {
```
Comment (markusbattarbee) [inline vlasovsolver/cpu_trans_map.cpp:660]: this comment looks like it should be removed

File: spatial_cell_cpu.cpp [hunk 4]
```diff
@@ -642,6 +652,13 @@ namespace spatial_cell {
             if ( P::amrMaxSpatialRefLevel == 0 || receiving || ranks.find(receiver_rank) != ranks.end()) {
                
                for ( int i = 0; i < MAX_NEIGHBORS_PER_DIM; ++i) {
+                  if(!receiving) {
+                     for(unsigned int j=0; j < this->neighbor_number_of_blocks[i]*VELOCITY_BLOCK_LENGTH; j++) {
```
Comment (markusbattarbee) [inline spatial_cell_cpu.cpp:656]: This should probably be deleted or go behind a debug ifdef

**Review by markusbattarbee — CHANGES_REQUESTED** [review]

**Comment by ursg** [comment]
Hmm, these testpackage outputs are a bit puzzling. Flowthrough_amr diffs make sense with smaller induced refinement, but those magnetospheres shouldn't probably be that different.
And what's with those Waitall errors? It looks like output still got written? Are they happening at the end of runs?

**Comment by ykempf** [comment]
> Hmm, these testpackage outputs are a bit puzzling. Flowthrough_amr diffs make sense with smaller induced refinement, but those magnetospheres shouldn't probably be that different. And what's with those Waitall errors? It looks like output still got written? Are they happening at the end of runs?

See my comments on Zulip...

File: grid.cpp [hunk 5]
```diff
@@ -994,24 +976,6 @@ void initializeStencils(dccrg::Dccrg<SpatialCell,dccrg::Cartesian_Geometry>& mpi
    // In spatial AMR using DCCRG, the neighbors are considered relative to a given cell's size.
    // To get two coarse neighbors from a fine cell at interfaces, the stencil size needs to be increased by one.
    int addStencilDepth = 0;
```
Comment (markusbattarbee) [inline grid.cpp:978]: addStencilDepth should be purged throughout

File: .gitmodules [hunk 6]
```diff
@@ -4,7 +4,7 @@
 
 [submodule "submodules/dccrg"]
 	path = submodules/dccrg
-	url = https://github.com/fmihpc/dccrg.git
+	url = https://github.com/ursg/dccrg.git
```
Comment (alhom) [inline .gitmodules:7]: Shouldn't the relevant commit be visible via the PR in fmihpc - no need to point to ursg/dccrg?

File: vlasovsolver/cpu_trans_map.cpp (hunk 1 above)
Comment (ursg) [reply vlasovsolver/cpu_trans_map.cpp:693]: done.

File: vlasovsolver/cpu_trans_map.cpp (hunk 3 above)
Comment (ursg) [reply vlasovsolver/cpu_trans_map.cpp]: Indeed. Removed.

File: grid.cpp (hunk 5 above)
Comment (ursg) [reply grid.cpp]: Done.

File: .gitmodules (hunk 6 above)
Comment (ursg) [reply .gitmodules]: Looks like that works indeed. Nice!

**Comment by ursg** [comment]
But actually, our fancy erase_if mechanism from DCCRG turns out not to even compile in the build_testpackage CI step!

**Comment by markusbattarbee** [comment]
Included in #1060

**Pull request closed without merging**